src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Metrics/SocketsHttpHandlerMetrics.cs

```diff
@@ -19,5 +20,42 @@ internal sealed class SocketsHttpHandlerMetrics(Meter meter)
             name: "http-client-connection-duration",
             unit: "s",
             description: "The duration of outbound HTTP connections.");
+
+        public readonly Histogram<double> RequestsQueueDuration = meter.CreateHistogram<double>(
+            name: "http-client-requests-queue-duration",
+            unit: "s",
+            description: "The amount of time requests spent on a queue waiting for an available connection.");
+
+        public void RequestLeftQueue(HttpConnectionPool pool, TimeSpan duration, int versionMajor)
+        {
+            Debug.Assert(versionMajor is 1 or 2 or 3);
+
+            if (RequestsQueueDuration.Enabled)
+            {
+                TagList tags = default;
+
+                tags.Add("protocol", versionMajor switch
+                {
+                    1 => "HTTP/1.1",
+                    2 => "HTTP/2",
+                    _ => "HTTP/3"
+                });
+
+                tags.Add("scheme", pool.IsSecure ? "https" : "http");
+                tags.Add("host", pool.OriginAuthority.HostValue);
+
+                if (!pool.IsDefaultPort)
+                {
+                    tags.Add("port", pool.OriginAuthority.Port);
+                }
+
+                RequestsQueueDuration.Record(duration.TotalSeconds, tags);
+            }
+
+            if (HttpTelemetry.Log.IsEnabled())
+            {
+                HttpTelemetry.Log.RequestLeftQueue(versionMajor, duration);
+            }
```

Comment:
Nit: maybe it's ok do dupe this out to callers so we don't do `HttpTelemetry` implicitly in metrics methods?